src/main/java/executor/command/CommandConvert.java

```diff
@@ -0,0 +1,194 @@
+        String fromStr = Parser.parseForFlag("from", userInput);
+        return fromStr;
+    }
+
+    /**
+     * getToUserInput parses the user input for the flag "to".
+     * @param userInput this is the user entered input from CLI
+     * @return this function return the 3 character unique string
+     */
+    private String getToUserInput(String userInput) {
+        String toStr = Parser.parseForFlag("to", userInput);
+        return toStr;
+    }
+
+    /**
+     * getTheJson is a function which gets the JSON object from the api.
+     * @param from this is the country from which we are converting currency
+     * @param to this is the country to which we are converting the currency
+     * @return this function returns a string version of the json object or else it will return null
+     */
+    private String getTheJson(String from, String to) {
+        try {
+            URL url = new URL("https://api.ratesapi.io/api/latest?base=" + from + "&symbols=" + to);
+            BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()));
+            String str = "";
+            String completeJson = "";
+            while (null != (str = br.readLine())) {
+                completeJson += str;
+            }
+            return completeJson;
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     * convertCurrency is a function used to loop through the string version of the json.
+     * @param from this is the country from which we are converting currency
+     * @param to this is the country to which we are converting the currency
+     * @param amount this is amount of currency which the user wants to convert
+     * @return this function returns the converted amount of the currency or else returns null
+     */
+    private Double convertCurrency(String from, String to, Double amount) {
+        try {
+            String json = getTheJson(from,to);
+            if (json != null) {
+                JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
+                String rate = jsonObject.getAsJsonObject("rates").get(to).getAsString();
+                BigDecimal exchangeRate = new BigDecimal(rate);
+                double exRate = exchangeRate.doubleValue();
+                setExchangeRate(exRate);
+                double convertedAmount = exRate * amount;
+                return convertedAmount;
+            }
+        } catch (Exception e) {
+            Ui.dukeSays(e.getMessage());
+            Ui.dukeSays(Ui.LINE);
+            Ui.dukeSays("Please enter in the following format : "
+                    + "convert 2000 /from USD /to EUR");
+        }
+        return null;
+    }
+
+
+    private String getFrom() {
+        return from;
+    }
+
+    private String getTo() {
+        return to;
+    }
+
+    private Double getAmount() {
+        return amount;
+    }
```

Comment:
Setters and Getters should be public/package-private or else there's no point of them existing. Group all Setters and Getters together at the bottom of the Class.

Reply:
Moved to the bottom of the class. I needed the getters because in the last part where i print to the ui, i call from the private variables stored within the class itself. Didnt want to make it public because, user is not allowed to manipulate these values.